lldb/source/Plugins/Language/CPlusPlus/CPlusPlusLanguage.cpp

```diff
@@ -236,199 +236,141 @@ static bool PrettyPrintFunctionNameWithArgs(Stream &out_stream,
   return true;
 }
 
-static std::optional<llvm::StringRef>
-GetDemangledBasename(const SymbolContext &sc) {
+static llvm::Expected<std::pair<llvm::StringRef, DemangledNameInfo>>
+GetAndValidateInfo(const SymbolContext &sc) {
   Mangled mangled = sc.GetPossiblyInlinedFunctionName();
   if (!mangled)
-    return std::nullopt;
+    return llvm::createStringError("Function does not have a mangled name.");
 
   auto demangled_name = mangled.GetDemangledName().GetStringRef();
   if (demangled_name.empty())
-    return std::nullopt;
+    return llvm::createStringError("Function does not have a demangled name.");
```

Comment:
```suggestion
    return llvm::createStringError("Function '%s' does not have a demangled name.", mangled.GetMangledName().AsCString(""));
```

Reply:
Added better messages for all the `createStringError`.